src/exchange/moderators/truncate.py

```diff
@@ -1,59 +1,70 @@
-from typing import Literal, Optional, Type
+from typing import List, Optional, Type
 
+from exchange.checkpoint import CheckpointData
+from exchange.message import Message
 from exchange.moderators import PassiveModerator
 from exchange.moderators.base import Moderator
 
-MAX_TOKENS = 112000
-
-
-def pop_checkpoint(
-    exchange: Type["exchange.exchange.Exchange"],  # noqa: F821
-    exclude_last: Literal[0, 1] = 0,
-) -> Type["exchange.exchange.Exchange"]:  # noqa: F821
-    """Pop messages from the front of the list in sections"""
-    checkpoint = exchange.checkpoints.pop(0)
-    for _ in range(checkpoint.end_index - checkpoint.start_index - exclude_last):
-        exchange.messages.pop(checkpoint.start_index)
-
-    # rewrite checkpoint indexes
-    for cp in exchange.checkpoints:
-        cp.start_index = cp.start_index - checkpoint.end_index + exclude_last
-        cp.end_index = cp.end_index - checkpoint.end_index + exclude_last
-
-    if exclude_last:
-        # there is still one entry in the first checkpoint only
-        checkpoint.token_count = checkpoint.latest_generated_tokens
-        checkpoint.end_index = exclude_last
-        exchange.checkpoints.insert(0, checkpoint)
-
-    return exchange
+# currently this is the point at which we start to truncate, so
+# so once we get to this token size the token count will exceed this
+# by a little bit
+MAX_TOKENS = 70000
```

Comment:
This should be larger. 128K is the standard context lengths, and we have a max limit on tool usage at 16000 (https://github.com/square/exchange/blob/ed3a6ab8ff4d414494980da4cfc3604264ec3e9b/src/exchange/exchange.py#L20), which is where we got the 112000 before. We'll start loosing too much context if we start at 70K.

Reply:
Awesome, thanks for explaining that! I'll drop some comments in the code for posterity.